application/controllers/admin/dumpdb.php

```diff
@@ -69,4 +69,38 @@ private function outputHeaders(string $sFileName)
         header('Content-Disposition: attachment; filename=' . $sFileName);
         header("Cache-Control: no-store, no-cache, must-revalidate");  // Don't store in cache because it is sensitive data
     }
+
+    private function getData()
+    {
+        Yii::app()->loadHelper("admin/backupdb");
+        $dbSize = getDatabaseSize();
+        $downloadable = true;
+        if ($dbSize > Yii::app()->getConfig('maxDatabaseSizeForDump')) {
+            $downloadable = false;
+        }
+        return [
+            'downloadable' => $downloadable,
+            'dbSize' => $dbSize,
+        ];
+    }
+
+    public function outPutDatabase()
+    {
+        // Check if it's a POST request
+        if (!Yii::app()->request->isPostRequest) {
+            throw new CHttpException(405, gT("Invalid action"));
+        }
+
+        // Check if user has necessary permissions
+        if (!Permission::model()->hasGlobalPermission('superadmin', 'read')) {
+            throw new CHttpException(403, gT("You do not have permission to access this page."));
+        }
+
```

Comment:
Need to add another check for the size of the db to prevent an admin to be able to download a db dump making a post request even if the page is requesting to schedule the database.